kview/src/main/java/dev/ikm/komet/kview/mvvm/model/DataModelHelper.java

```diff
@@ -77,4 +104,90 @@ public static Set<ConceptEntity> fetchDescendentsOfConcept(ViewProperties viewPr
                 .collect(Collectors.toSet());
         return allDecendents;
     }
+
+    public static List<PatternEntityVersion> getMembershipPatterns() {
+        List<PatternEntityVersion> membershipPatternList = new ArrayList<>();
+        PrimitiveData.get().forEachPatternNid((patternNid) -> {
+            PatternEntityVersion patternEntityVersion = (PatternEntityVersion)
+                    Calculators.View.Default().stampCalculator().latest(patternNid).get();
+            if (patternEntityVersion.semanticPurposeNid() == TinkarTerm.MEMBERSHIP_SEMANTIC.nid()) {
+                membershipPatternList.add(patternEntityVersion);
+            }
+        });
+        return membershipPatternList;
+    }
+
+    public static boolean isInMembershipPattern(int conceptNid, int patternNid) {
+        int[] semanticCount = PrimitiveData.get().semanticNidsForComponentOfPattern(conceptNid, patternNid);
+        // if the semantic count is empty then it is not a member of that pattern
+        return semanticCount.length > 0;
+    }
+
+    public static void addToMembershipPattern(EntityFacade concept, EntityFacade pattern, ViewCalculator viewCalculator) {
+        EditCoordinate editCoordinate = viewCalculator.viewCoordinateRecord().editCoordinate();
+        int[] semanticNidsForComponent = PrimitiveData.get().semanticNidsForComponentOfPattern(concept.nid(), pattern.nid());
+        if (semanticNidsForComponent.length == 0) {
+            // case 1: never a member
+            createSemantic(concept, pattern, editCoordinate.toEditCoordinateRecord(), viewCalculator);
+        } else {
+            // a member, need to change to inactive.
+            updateSemantic(pattern, semanticNidsForComponent[0], editCoordinate.toEditCoordinateRecord(), viewCalculator);
+        }
+    }
+
+    public static void removeFromMembershipPattern(EntityFacade concept, EntityFacade pattern, ViewCalculator viewCalculator) {
```

Comment:
Can the first two arguments be a PublicId or nid? 
Can we document the method?

Reply:
I need to call Pattern.toProxy() in updateSemantic() which this method calls... for this method the concept's entityFacade argument can be replaced by it's nid since it doesn't look like we need anything else from the EntityFacade down the call stack

Reply:
updated first arg to be a nid